src/AcceptanceTests/Shared/Retry.cs

```diff
@@ -22,51 +23,112 @@ public async Task Should_work()
             {
-                Assert.That(receivedHeaderValue, Is.EqualTo(header.Value),
+                if (translateReplyToAdressForFailedMessages && header.Key == Headers.ReplyToAddress)
+                {
+                    Assert.That(receivedHeaderValue.Contains(nameof(ProcessingEndpoint), StringComparison.InvariantCultureIgnoreCase), Is.True,
+                        $"The ReplyToAddress received by ServiceControl ({TransportBeingTested} physical address) should contain the logical name of the endpoint.");
+                }
+                else
+                {
+                    Assert.That(receivedHeaderValue, Is.EqualTo(header.Value),
                     $"{header.Key} is not the same on processed message and message sent to the error queue");
+                }
             }
         }
     }
 
+    [Test]
+    public async Task Should_log_warn_when_best_effort_ReplyToAddress_fails()
+    {
+        var ctx = await Scenario.Define<Context>()
+            .WithEndpoint<SendingEndpoint>(builder =>
+            {
+                builder.DoNotFailOnErrorMessages();
+                builder.When(c => c.EndpointsStarted, (session, _) => session.Send(new FaultyMessage()));
+            })
+            .WithEndpoint<ProcessingEndpoint>(builder =>
+            {
+                builder.DoNotFailOnErrorMessages();
+                builder.When(c => c.EndpointsStarted, (session, _) =>
+                {
+                    var options = new SendOptions();
+                    options.RouteToThisEndpoint();
+                    options.SetHeader(Headers.ReplyToAddress, "address-not-declared-in-the-bridge");
+                    return session.Send(new FaultyMessage(), options);
+                });
+            })
+            .WithEndpoint<FakeSCError>()
+            .WithBridge(bridgeConfiguration =>
+            {
+                bridgeConfiguration.TranslateReplyToAddressForFailedMessages();
+                var bridgeTransport = new TestableBridgeTransport(DefaultTestServer.GetTestTransportDefinition())
+                {
+                    Name = "DefaultTestingTransport"
+                };
+                bridgeTransport.AddTestEndpoint<FakeSCError>();
+                bridgeConfiguration.AddTransport(bridgeTransport);
+
+                var theOtherTransport = new TestableBridgeTransport(TransportBeingTested);
+                theOtherTransport.AddTestEndpoint<ProcessingEndpoint>();
+                bridgeConfiguration.AddTransport(theOtherTransport);
+            })
+            .Done(c => c.GotRetrySuccessfullAck)
+            .Run();
+
+        var translationFailureLogs = ctx.Logs.ToArray().Where(i =>
+            i.Message.Contains("Could not translate") &&
+            i.Message.Contains("address. Consider using `.HasEndpoint()`"));
+
+        //There is only one warning here because the ServiceControl testing fake does not properly set the ReplyToAddress header value
```

Comment:
Should we make the fake set the value?

Reply:
Not sure that it would add any value to what we are testing?

Reply:
Fair enough 👍